docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,491 @@
+    Fleet_Paused--> Individual_Device: no matching fleet labels
+```
+
+##### `/device/{name}` with no owner fleet (Individual Device)
+
+Devices with no matching labels to a fleet are manually managed and start with no `metadata.owner`
+reference the fleet controller will not attempt to reconcile or configure the device in any way.
+
+A `metadata.owner` field can be written through the API to associate the device to an external
+owner (like ACM or AAP, etc..), and prevent the fleet controller from managing the device in any way.
+
+If a fleet matches this device, and the device has empty `metadata.owner` the fleet controller
+will claim the device, by setting `metadata.owner` to `Fleet/{fleet-name}`.
+
+The `os.image` and configuration fields in this API support floating tags, and while the fleet manager
+does freeze the tags to hashes, the user is free to setup the device spec with floating references.
+
+###### Expected behaviour regarding external references with floating tags
+
+Changes on external references are monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes to the rendered version which will be exposed
+on the `/device/{name}/rendered` endpoint.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+##### `/device/{name}` while managed by a fleet (Fleet Device)
+
+When managing a device through a fleet, flightctl is responsible
+for maintaining the device spec in sync with the fleet spec; in this case
+*write access is only possible for the modification of labels and other metadata*
+through this API except for the owner fleet.
+
+> [!NOTE]
+> Devices managed by a fleet have their floating tags resolved to stable hashes at this API
+> level. This is done to ensure that the device is always in sync with the fleet spec.
+
+The device will have an owner reference to the managing fleet, i.e.:
+
+```yaml
+  owner: Fleet/optical-inspector-production-fleet
+```
+
+###### Temporarily disabling fleet management (Fleet Device paused)
+
+A user can temporarily stop fleet management of a device by setting the `flightctl.io/fleet-controller=paused`
+in the device labels, the fleet manager will ignore devices with this label for any subsequent
+update or rollout.
+
+This feature is intended to allow temporarily debugging a device without de-associating
+from the fleet. This will mean the `metadata.owner` field is not cleared. 
+
+Devices set to `flightctl.io/fleet-controller=paused` with a `metadata.owner` starting
+with "Fleet/" must signal a device status condition to indicate that the device is not
+being synchronized to the owner fleet.
+
+```yaml
+  - lastTransitionTime: "2024-04-30T15:06:20Z"
+    message: This device is not in sync with the owner fleet.
+    reason: Management Paused from flightctl.io/fleet-controller=paused label.
+    status: False
+    type: Management
```

Comment:
This would mean the condition is part of the device spec, but device don't know anything about fleets...

Reply:
Ok, I will remove it, we would assume it from the label being there.

Reply:
This makes me wonder if we need some mechanisms to let controllers setup status conditions on devices, may be via a different status field (controllerStatus), and a different endpoint?

I suspect we will need to figure out something for that eventually, because even though we don't want the devices to have knowledge of the controller, there is some status relative to the controller that needs to go into devices (this will be still more evident if we have templating, and suddenly a device cannot be rendered by a template...)